homeassistant/components/overkiz/water_heater_entities/domestic_hot_water_production.py

```diff
@@ -0,0 +1,281 @@
+"""Support for DomesticHotWaterProduction."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.water_heater import (
+    STATE_ECO,
+    STATE_HIGH_DEMAND,
+    STATE_OFF,
+    STATE_PERFORMANCE,
+    WaterHeaterEntity,
+    WaterHeaterEntityFeature,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..entity import OverkizEntity
+
+OVERKIZ_TO_OPERATION_MODE: dict[str, str] = {
+    OverkizCommandParam.STANDARD: STATE_ECO,
+    OverkizCommandParam.HIGH_DEMAND: STATE_HIGH_DEMAND,
+    OverkizCommandParam.STOP: STATE_OFF,
+    OverkizCommandParam.MANUAL_ECO_ACTIVE: STATE_ECO,
+    OverkizCommandParam.MANUAL_ECO_INACTIVE: STATE_OFF,
+    OverkizCommandParam.AUTO: STATE_ECO,
+    OverkizCommandParam.AUTO_MODE: STATE_ECO,
+    OverkizCommandParam.BOOST: STATE_PERFORMANCE,
+}
+
+OPERATION_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_OPERATION_MODE.items()}
+
+DHWP_AWAY_MODES = [
+    OverkizCommandParam.ABSENCE,
+    OverkizCommandParam.AWAY,
+    OverkizCommandParam.FROSTPROTECTION,
+]
+
+DHWP_BOOST_MODES = [
+    OverkizCommandParam.BOOST,
+]
```

Comment:
Why a list for one value ?

Reply:
For 2 reasons:
1. To have the same code for boost or away modes
2. To ease the add of new boost modes of a new device if needed

This entity is able to handle a lot of different devices.
I only find 3 different setup for a least 12 differnts devices.
I'm pretty sure they're will be devices with other boost mode aliases.

Let me know if you want me to revert it and use a simple string for now

Reply:
I understand your vision of preventing future evolution or fixes, but I'm afraid we are not doing this here.

A PR should serve one purpose of a current issue or feature.

Reply:
Fixed.